src/web/js/output-ui.js

```diff
@@ -612,70 +394,46 @@
       });
     }
     
-    function getSourceContent(editors, cmloc, tight) {
-      if(editors.hasOwnProperty(cmloc.source)) {
-        if(!tight) {
-          cmloc =
-            {start:{
-              line: cmloc.start.line,
-              ch: 0},
-             end:{
-              line: cmloc.end.line,
-              ch: editors[cmloc.source].getLine(cmloc.end.line).length}};
-        }
-        return editors[cmloc.source].getRange(cmloc.start, cmloc.end);
-      } else {
-        var source = sessionStorage.getItem(cmloc.source);
-        if(!source) {
-          return undefined;
-        }
-        var lines = source.split("\n").slice(cmloc.start.line, cmloc.end.line + 1);
-        if(tight) {
-          lines[lines.length - 1] = lines[lines.length - 1].substr(0, cmloc.end.ch);
-          lines[0] = lines[0].substr(cmloc.start.ch);
-        }
-        return lines.join("\n");
-      }
-    }
-    
-    function makeMaybeLocToAST(runtime, editors, srcloc) {
+    function makeMaybeLocToAST(runtime, documents, srcloc) {
       return runtime.makeFunction(function(loc) {
         return runtime.ffi.cases(runtime.getField(srcloc, "is-Srcloc"), "Srcloc", loc, {
           "builtin": function(_) {
             console.error("maybeLocToAST should not be passed a builtin source location.", loc);
             return runtime.ffi.makeNone();
           },
-          "srcloc": function(filename, start_line, start_col, startCh, endL, endC, endCh) {
+          "srcloc": function(filename, start_line, start_col, _, end_line, end_col, _) {
```

Comment:
@jpolitz Addressed by 4f699ab746a209288ab1def6bca87ad5d2b1c640. Some of the changes get a little silly, such as:
```
function(filename, _, __, ___, ____, _____, ______) { 
  ...
```

It would be valid javascript to take advantage of loosey-goosy arity and write
```
function(filename) { 
  ...
```
but that seems like playing with fire.